### DISCUSSION TITLE
Using the Config Context/Templates, how am I meant to get interface information?

### DISCUSSION THREAD

**Discussion by WhaleJ84** [netbox-community/netbox#17816] [Q&A] [answered]
Hey, I have a device that has 6 interfaces, with IP addresses assigned to each. Using a for loop, I am trying to render each interface in an IOS configuration manner in the config context. However trying to access the information using:
```
{% for interface in device.interfaces.all %}
  {{ interface.name }}
{% endfor %}
```
or
```
{% for interface in device.interfaces %}
  {{ interface.name }}
{% endfor %}
```
either returns a TypeError ('method' object is not iterable) for the former, or  TypeError ('RelatedManager' object is not iterable) for the latter.

Is this a bug in the API - not returning any interface information for the device, a missing feature, or am I doing this the wrong way?
I am running Netbox 4.1.4 and can provide more information if needed.

Thanks in advance.

**Comment by pl0xym0r** [comment, marked as answer]
device.interfaces.all()

**Reply by WhaleJ84** [reply]
Of course it was a missing `()` - thanks!
